### PR TITLE
modemmanager: bump to 1.22.0

### PR DESCRIPTION
**Pull request by Linaro1985**
Update modemmanager to 1.22.0 and necessary dependencies (libmbim and libqmi). 

- switch to builtin plugins.
- import upstream patch for fixing support of ublox LARA-R6001 / LARA-R6001D modules
  (see: https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/issues/779)

Signed-off-by: Maxim Anisimov [maxim.anisimov.ua@gmail.com](mailto:maxim.anisimov.ua@gmail.com)

Maintainer: Nicholas Smith [nicholas@nbembedded.com](mailto:nicholas@nbembedded.com)

Compile tested:
on amd64 for mipsel_24kc on https://git.openwrt.org/openwrt/openwrt.git commit https://github.com/openwrt/packages/commit/7ed86d2eb7a58a6ed938b08fcacb4461a2876631

Run tested:
Zbtlink ZBT-WG3526

### CODE REVIEW

File: net/modemmanager/Makefile [hunk 1]
```diff
@@ -64,6 +64,7 @@ MESON_ARGS += \
 	-Dintrospection=false \
 	-Dman=false \
 	-Dbash_completion=false \
+	-Dbuiltin_plugins=true \
```
Comment (feckert) [inline net/modemmanager/Makefile:67]: I don't like the idea of including the plugins in the ModemManager binary. All plugins, whether they are needed or not, have to be loaded into RAM when the ModemManager is started. I think this is currently not the case for plugins that are not needed at runtime.

**Review by feckert — CHANGES_REQUESTED** [review]
Also the ModemManager compilesl, but does not work.
Have a look at the [issue](https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/issues/779) I have opened on the ModemManager page

File: net/modemmanager/Makefile (hunk 1 above)
Comment (Linaro1985) [reply net/modemmanager/Makefile:67]: The modemmanager author @aleksander0m wrote that this option works faster. Also, my total package size has become smaller.

**Comment by Linaro1985** [comment]
> Also the ModemManager compilesl, but does not work

I have a Quectel EP06 modem. It worked on it. I'll look for other modems to test.

File: net/modemmanager/Makefile (hunk 1 above)
Comment (aleksander0m) [reply net/modemmanager/Makefile:67]: All plugins whether needed or not we're being loaded at runtime. Including all in the MM binary is much better, if there is no plan to allow third party plugins

Comment (feckert) [reply net/modemmanager/Makefile:67]: Up to now, I think there is no plan to support third party plugins in openwrt. So I think this is OK. But I am little surprised, that the ModemManager is realy smaller, if the plugins are all build in! I would not have expect that.

**Review by feckert — APPROVED** [review]

**Comment by feckert** [comment]
Thanks merged!

**Comment by feckert** [comment]
@BKPepe   I know the first openwrt-23.05 was release last week, And I know that updating a package in a release is not desired, but the ModemManager  1.22 version was released 1 day after openwrt-23.05 release. Its not a major ModemManager update, but it has some nice new features and bug fixes.

**Comment by BKPepe** [comment]
Go ahead! :)

**Pull request merged**